web/yo/app/scripts/controllers/gene.js

```diff
@@ -3028,33 +3020,11 @@ angular.module('oncokbApp')
             }
             function loadMetaFile(callback) {
                 if(!$rootScope.metaData) {
-                    storage.retrieveMeta().then(function(result) {
-                        if (result && (result.error || !_.isArray(result) || result.length === 0)) {
-                            dialogs.error('Error', 'Fail to retrieve meta file! Please stop editing and contact the developer!');
-                            var sendTo = 'dev.oncokb@gmail.com';
-                            var subject = 'Fail to retrieve meta file';
-                            var content;
-                            if(_.isArray(result) && result.length === 0) {
-                                content = 'There is no meta file inside the Meta folder';
-                            } else {
-                                content = 'System error is ' + JSON.stringify(result.error);
-                            }
-                            mainUtils.sendEmail(sendTo, subject, content);
-                            callback();
-                        } else {
-                            storage.getMetaRealtimeDocument(result[0].id).then(function(metaRealtime) {
-                                if (metaRealtime && metaRealtime.error) {
-                                    dialogs.error('Error', 'Fail to get meta document! Please stop editing and contact the developer!');
-                                    $scope.fileEditable = false;
-                                    callback();
-                                } else {
-                                    $rootScope.metaRealtime = metaRealtime;
-                                    $rootScope.metaModel = metaRealtime.getModel();
-                                    $rootScope.metaData = metaRealtime.getModel().getRoot().get('review');
-                                    assignMeta(callback);
-                                }
-                            });
-                        }
+                    additionalFile.load(['all']).then(function(result) {
+                        assignMeta(callback);
+                    }, function(error) {
+                        $scope.fileEditable = false;
+                        callback();
```

Comment:
Usually, we do check on callback whether it is a function. But this function is ok since it is a private function and will only be used in gene.js. But for all functions in utils should be checked.